Index: fizz/fizz/CMakeLists.txt
===================================================================
--- fizz.orig/fizz/CMakeLists.txt
+++ fizz/fizz/CMakeLists.txt
@@ -62,8 +62,8 @@ endif()
 
 include(CheckAtomic)
 
-find_package(Sodium REQUIRED)
-set(FIZZ_HAVE_SODIUM ${Sodium_FOUND})
+find_package(LibSodium REQUIRED)
+set(FIZZ_HAVE_SODIUM ${LibSodium_FOUND})
 
 SET(FIZZ_SHINY_DEPENDENCIES "")
 SET(FIZZ_LINK_LIBRARIES "")
@@ -201,7 +201,7 @@ target_include_directories(
     $<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>
     ${FOLLY_INCLUDE_DIR}
     ${OPENSSL_INCLUDE_DIR}
-    ${sodium_INCLUDE_DIR}
+    ${LIBSODIUM_INCLUDE_DIR}
     ${ZSTD_INCLUDE_DIR}
   PRIVATE
     ${GLOG_INCLUDE_DIRS}
@@ -213,7 +213,7 @@ target_link_libraries(fizz
   PUBLIC
     ${FOLLY_LIBRARIES}
     ${OPENSSL_LIBRARIES}
-    sodium
+    ${LIBSODIUM_LIBRARIES}
     Threads::Threads
     ZLIB::ZLIB
     ${ZSTD_LIBRARY}

