diff --git a/CMakeLists.txt b/CMakeLists.txt
index a57c45c67d2..2eac1845f97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -227,14 +227,14 @@ add_fbthrift_cpp_library(
   mcrouter_common_thrift
   mcrouter/lib/network/gen/Common.thrift
   DEPENDS mcrouter_carbon_thrift
-  OPTIONS stack_arguments sync_methods_return_try deprecated_terse_writes
+  OPTIONS stack_arguments sync_methods_return_try
 )
 
 add_fbthrift_cpp_library(
   mcrouter_memcache_thrift
   mcrouter/lib/network/gen/Memcache.thrift
   DEPENDS mcrouter_common_thrift
-  OPTIONS stack_arguments sync_methods_return_try deprecated_terse_writes
+  OPTIONS stack_arguments sync_methods_return_try
 )
 
 add_fbthrift_cpp_library(
@@ -242,7 +242,7 @@ add_fbthrift_cpp_library(
   mcrouter/lib/network/gen/MemcacheService.thrift
   SERVICES Memcache
   DEPENDS mcrouter_memcache_thrift
-  OPTIONS stack_arguments sync_methods_return_try deprecated_terse_writes
+  OPTIONS stack_arguments sync_methods_return_try
 )
 
 # Add source include directories to thrift targets
@@ -270,7 +270,9 @@ set(MCROUTER_LIB_SOURCES
   mcrouter/lib/Clocks.cpp
   mcrouter/lib/Compression.cpp
   mcrouter/lib/CompressionCodecManager.cpp
+  mcrouter/lib/DynamicUtil.cpp
   mcrouter/lib/FailoverErrorsSettingsBase.cpp
+  mcrouter/lib/FiberLocalInternal.cpp
   mcrouter/lib/IOBufUtil.cpp
   mcrouter/lib/IovecCursor.cpp
   mcrouter/lib/Lz4CompressionCodec.cpp
@@ -293,6 +295,7 @@ set(MCROUTER_LIB_SOURCES
   mcrouter/lib/carbon/ReplyCommon.cpp
   mcrouter/lib/carbon/RequestCommon.cpp
   mcrouter/lib/carbon/Result.cpp
+  mcrouter/lib/carbon/connection/ExternalCarbonConnectionImpl.cpp
 
   # Config
   mcrouter/lib/config/ConfigPreprocessor.cpp
@@ -353,6 +356,12 @@ set(MCROUTER_LIB_SOURCES
   mcrouter/lib/network/gen/CommonMessagesThrift.cpp
   mcrouter/lib/network/gen/MemcacheMessages.cpp
   mcrouter/lib/network/gen/MemcacheMessagesThrift.cpp
+  mcrouter/lib/network/gen/MemcacheRouterInfo.cpp
+  mcrouter/lib/network/gen/MemcacheRouterInfo-AllFastestRoute.cpp
+  mcrouter/lib/network/gen/MemcacheRouterInfo-BuildExtraProvider.cpp
+  mcrouter/lib/network/gen/MemcacheRouterInfo-ExternTemplate.cpp
+  mcrouter/lib/network/gen/MemcacheRouterInfo-FailoverRoute.cpp
+  mcrouter/lib/network/gen/MemcacheRouterInfo-HashRoute.cpp
 )
 
 add_library(mcrouter_lib ${MCROUTER_LIB_SOURCES})
@@ -404,6 +413,7 @@ set(MCROUTER_CORE_SOURCES
   mcrouter/ProxyStats.cpp
   mcrouter/route.cpp
   mcrouter/RoutingPrefix.cpp
+  mcrouter/Server.cpp
   mcrouter/ServiceInfo.cpp
   mcrouter/stats.cpp
   mcrouter/ThreadUtil.cpp
-- 
2.34.1
